core/src/main/java/com/instaclustr/esop/cli/Esop.java

```diff
@@ -92,7 +94,7 @@ protected void configure() {
         modules.add(new ExecutorsModule());
         modules.add(new UploadingModule());
         modules.add(new DownloadingModule());
-        modules.add(new HashModule(hashSpec));
+        modules.add(new HashModule(hashSpec, request.concurrentConnections));
```

Comment:
can't you put concurrent connections into the constructor of HashSpec? Just pass it to hashSpec so you do not need to accommodate the code by passing request everywhere.